llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -544,11 +544,12 @@ StringRef InstrProfSymtab::getCanonicalName(StringRef PGOName) {
 }
 
 Error InstrProfSymtab::addFuncWithName(Function &F, StringRef PGOFuncName) {
-  StringSet<> Names;
-  Names.insert(PGOFuncName);
-  Names.insert(InstrProfSymtab::getCanonicalName(PGOFuncName));
-  for (const auto &NameEntry : Names) {
-    StringRef Name = NameEntry.getKey();
+  SmallVector<StringRef, 2> Names;
+  Names.push_back(PGOFuncName);
+  StringRef CanonicalFuncName = getCanonicalName(PGOFuncName);
+  if (CanonicalFuncName != PGOFuncName)
```

Comment:
Since you are doing this check explicitly here, rather than add to a vector and iterating it, suggest putting the handling currently within the loop body into a lambda and simply calling it twice (with the second call guarded by this if).